### PR TITLE
Fix bad weakrefs and missing signal handler for singulos

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

This was literally being used as a getcomponent
Narsie still has this but thats for later
yes this is tested

### CODE REVIEW

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -1097,3 +1097,6 @@
 #define COMSIG_AQUARIUM_BEFORE_INSERT_CHECK "aquarium_about_to_be_inserted"
 #define COMSIG_AQUARIUM_SURFACE_CHANGED "aquarium_surface_changed"
 #define COMSIG_AQUARIUM_FLUID_CHANGED "aquarium_fluid_changed"
+
+//Singularity component signals
+#define COMSIG_SINGULO_SIZE_CHANGE "singulos_size_changed"
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals.dm:1102]: No. We're pretty strict about not having signals that are essentially just procs. If there is nothing other than the singularity itself that hooks to this, then it should not be a signal. This exact conversation was had on the PR itself.

File: code/datums/components/singularity.dm [hunk 2]
```diff
@@ -61,6 +62,8 @@
 	src.notify_admins = notify_admins
 	src.roaming = roaming
 	src.singularity_size = singularity_size
+	if(immediate_target)
+		target = immediate_target
```
Comment (Mothblocks) [inline code/datums/components/singularity.dm:66]: You're not hooking to when this is deleted, this is going to hard delete. *That is why it was a weakref in the first place*. Weakrefs are for non-owning references. This is a non-owning reference.

**Comment by Mothblocks** [comment]
This is not the direction for signals to take.

This is my component, so I'm a bit biased on this, but we have nearly 100 open PRs right now. Feel free to get someone else to reopen if they are interested in merging.

File: code/datums/components/singularity.dm (hunk 2 above)
Comment (TiviPlus) [reply code/datums/components/singularity.dm:66]: This was a hard reference to the target, I changed nothing about target handling except allowing it to be passed in addcomponent
This part is your mistake but id be willing to fix it

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (TiviPlus) [reply code/__DEFINES/dcs/signals.dm:1102]: @Mothblocks I dont see that conversation when looking there? I only see you being told to do actually use signals instead like I did here by spooky: https://github.com/tgstation/tgstation/pull/55096#discussion_r529380267

**Pull request closed without merging**